fe1-web/model/network/method/message/data/MessageData.ts

```diff
@@ -37,16 +37,39 @@ export interface MessageData {
 }
 
 /**
- * Returns if the message data is signed with token or not. If not, it has to be signed with
- * a public key.
+ * Map to know if the type of message we're going to send needs to be signed using a token, or
+ * the user's public key.
  *
  * @remarks
- * This function has to be updated for each new kind of message that is going to be signed using
- * the pop token.
- *
- * @param data - The message data we want to know how to sign
+ * This map has to be updated for each new kind of message.
  */
-export function isSignedWithToken(data: MessageData): Boolean {
-  return (data.object === ObjectType.ELECTION && data.action === ActionType.CAST_VOTE)
-    || (data.object === ObjectType.CHIRP && data.action === ActionType.ADD);
-}
+export const isMessageSignedWithToken: Map<string, Map<string, boolean>> = new Map([
```

Comment:
as discussed on slack, I'd suggest using an interface rather than a `boolean` as the content of this map of maps.
it'd make your life easier when it comes to adding more elements.

Reply:
I tried to do it, but I'm not sure about what I've done.